### DISCUSSION TITLE
Emoji’s in chat!

### DISCUSSION THREAD

**Discussion by ablac** [Zrips/CMI#4061] [Feature Requests]
**Description of feature/improvement request:**

I was looking around and there’s a few plugins that add emojis’s in chat, Notably EmojiChat does this the best, as it has a Rudimentary GUI that allows somebody to select the Emoji/see the code to run it.

In CMI using a GUI similar to /colorpicker would make this so much more efficient, and amazing! Allowing users to do /emoji and it opens up a GUI similar to colorpicker with all the available Emoji , and gives them recipe for the emoji. Ex: :) or :-) or %Smile%

I feel this would be a great addition to the CMI chat functions.

Edit: I do believe this could be done with JSON Fonts as of 1.16.x instead of a resource pack as emojichat is doing this also.

---

### Basic explanation as to why this feature is needed and/or useful 

It’s not that it’s needed, but more of an awesome feature that’s rare, and would make somebody who’s running CMI have there server stand out, and get people questioning how it’s possible!

**Comment by CloudeLecaw** [comment]
You can use chatfilter for this an replace it with the smilies. We do this on our server too to have over 30 smilies in chat


Take a look here ;)
https://github.com/Zrips/CMI/issues/3995

For picking the emoji u can create a ctext where you print oit all smilies and make them past to chat on shift + click

Then create a customalias that calls thia ctext for the player
Then they can do /emojie to het the ctext that you desinged

You see, your request can allready be done by cmi :) 

For colored smilie you need to allow your player colorcodes thwn they can use them before the smilies

**Comment by ablac** [comment]
Yes, I do understand that, but with recent changes to Minecraft 1.16.1 should allow for a full fledged Emoji system like you can find on your phone. I know about the limited selection of Smilies that you can copy and paste in, but I’d really like to be able to do the full range of Smilies like in the plugin I posted in my OP, but using the new abilities in MC 1.16 without the resource pack.
